src/core/immutableMap.ts

```diff
@@ -8,7 +8,10 @@ type MCreate = <K, V>() => ImmutableMap<K, V>
 type MGet = <K, V>(m: ImmutableMap<K, V>, k: K) => V | undefined
 type MSet = <K, V>(m: ImmutableMap<K, V>, k: K, v: V) => ImmutableMap<K, V>
 type MDel = <K, V>(m: ImmutableMap<K, V>, k: K) => ImmutableMap<K, V>
-type MKeys = <K, V>(m: ImmutableMap<K, V>) => Set<K>
+type MForEach = <K extends object, V>(
+  m: ImmutableMap<K, V>,
+  cb: (value: V, key: K, map: Map<K, V>) => void
```

Comment:
My bad, I didn't mean to add the third argument, because it's not something we expect.
```suggestion
  cb: (value: V, key: K) => void
```
Let's do this.